### PR TITLE
Configurate spring in project

### PR DESCRIPTION
**Pull request by floriangeyer21**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,6 +50,57 @@
...
+            <artifactId>spring-web</artifactId>
+            <version>5.2.2.RELEASE</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <version>4.0.1</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet.jsp</groupId>
```
Comment (KovalAlexey) [inline pom.xml]: we don't use views here, so you can remove redundant dependencies like jsp and jstl

File: src/main/java/com/movie/store/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
 import com.movie.store.service.ShoppingCartService;
 import com.movie.store.service.UserService;
 import com.movie.store.util.HashUtil;
 import java.util.Optional;
+import org.springframework.stereotype.Service;
 
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
-    @Inject
     private UserService userService;
```
Comment (KovalAlexey) [inline src/main/java/com/movie/store/security/AuthenticationServiceImpl.java]:
```suggestion
    private final UserService userService;
```

Comment (KovalAlexey) [reply src/main/java/com/movie/store/security/AuthenticationServiceImpl.java]: same though the project

**Review by KovalAlexey — APPROVED** [review]

File: pom.xml [hunk 3]
```diff
@@ -49,6 +50,57 @@
...
+            <artifactId>javax.servlet.jsp-api</artifactId>
+            <version>2.3.1</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet.jsp.jstl</groupId>
```
Comment (RomanDubovskyi) [inline pom.xml]: you don't use jsp, let's remove these dependencies

File: src/main/java/com/movie/store/dao/impl/CinemaHallDaoImpl.java [hunk 4]
```diff
@@ -2,25 +2,33 @@
...
 import javax.persistence.criteria.CriteriaQuery;
+import lombok.extern.log4j.Log4j;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+@Log4j
+@Repository
+public class CinemaHallDaoImpl extends AbstractSessionFactoryCreator implements CinemaHallDao {
```
Comment (RomanDubovskyi) [inline src/main/java/com/movie/store/dao/impl/CinemaHallDaoImpl.java]: it's not logical that Dao extends SessionFactoryCreator + you still call constructor in each class, so class AbstractSessionFactoryCreator is redundant, let's remove it

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**